python/mxnet/gluon/nn/basic_layers.py

```diff
@@ -62,7 +62,15 @@ def __repr__(self):
                         modstr=modstr)
 
     def __getitem__(self, key):
-        return list(self._children.values())[key]
+        layers = list(self._children.values())[key]
+        if isinstance(layers, list):
+            with self.name_scope():
+                net = type(self)()
```

Comment:
why type(self)? There is no guarantee that type(self)'s constructor accepts no arguments

Reply:
This is so that any blocks that inherit sequential blocks have this default implementation. Child class that have other arguments can choose to override this.